File: py/packages/halo/src/halo_format/node.py
"""The node model and its (de)serialization.

    BranchNode = {"k": "b", "summary": str, "branches": {name: Handle}}
    LeafNode   = {"k": "l", "value": JsonValue}

The k kind tag is part of the hashed content so a leaf and branch can never collide. Stored form
is canonical(node) bytes, not the object.
"""

File: ts/packages/halo/src/node.ts
// node: the node model and its (de)serialization.
//
//   type BranchNode = { k: "b"; summary: string; branches: Record<string, Handle> }
//   type LeafNode   = { k: "l"; value: JsonValue }
//   type Node       = BranchNode | LeafNode
//
// The `k` kind tag is part of the hashed content so a leaf and branch can never collide.
// Stored form is canonical(node) bytes, not the object.

export {};
